server/src/main/java/io/deephaven/server/auth/CommunityAuthenticationModule.java

```diff
@@ -0,0 +1,124 @@
+import io.deephaven.proto.backplane.grpc.ObjectServiceGrpc;
+import io.deephaven.proto.backplane.grpc.PartitionedTableServiceGrpc;
+import io.deephaven.proto.backplane.grpc.SessionServiceGrpc;
+import io.deephaven.proto.backplane.grpc.StorageServiceGrpc;
+import io.deephaven.proto.backplane.script.grpc.ConsoleServiceGrpc;
+import io.deephaven.server.session.TicketResolverBase;
+import io.grpc.health.v1.HealthGrpc;
+import org.apache.arrow.flight.impl.FlightServiceGrpc;
+
+import javax.inject.Singleton;
+
+@MapKey
+@interface ServiceGrpcKey {
+    String value();
+}
+
+
+@Module
+public class CommunityAuthenticationModule {
+    @Provides
+    @IntoMap
+    @ServiceGrpcKey(ApplicationServiceGrpc.SERVICE_NAME)
+    public ServiceAuthWiring bindApplicationServiceAuthWiring() {
+        return new ApplicationServiceAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @IntoMap
+    @ServiceGrpcKey(ConfigServiceGrpc.SERVICE_NAME)
+    public ServiceAuthWiring bindConfigServiceAuthWiring() {
+        return new ConfigServiceAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @IntoMap
+    @ServiceGrpcKey(ConsoleServiceGrpc.SERVICE_NAME)
+    public ServiceAuthWiring bindConsoleServiceAuthWiring() {
+        return new ConsoleServiceAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @IntoMap
+    @ServiceGrpcKey(ObjectServiceGrpc.SERVICE_NAME)
+    public ServiceAuthWiring bindObjectServiceAuthWiring() {
+        return new ObjectServiceAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @IntoMap
+    @ServiceGrpcKey(PartitionedTableServiceGrpc.SERVICE_NAME)
+    public ServiceAuthWiring bindPartitionedTableServiceAuthWiring() {
+        return new PartitionedTableServiceAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @IntoMap
+    @ServiceGrpcKey(SessionServiceGrpc.SERVICE_NAME)
+    public ServiceAuthWiring bindSessionServiceAuthWiring() {
+        return new SessionServiceAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @IntoMap
+    @ServiceGrpcKey(StorageServiceGrpc.SERVICE_NAME)
+    public ServiceAuthWiring bindStorageServiceAuthWiring() {
+        return new StorageServiceAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @IntoMap
+    @ServiceGrpcKey(FlightServiceGrpc.SERVICE_NAME)
+    public ServiceAuthWiring bindFlightServiceAuthWiring() {
+        return new FlightServiceAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @IntoMap
+    @ServiceGrpcKey(HealthGrpc.SERVICE_NAME)
+    public ServiceAuthWiring bindHealthAuthWiring() {
+        return new HealthAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @Singleton
+    public TableServiceContextualAuthWiring bindTableServiceContextualAuthWiring() {
+        return new TableServiceContextualAuthWiring.AllowAll();
+    }
+
+    @Provides
+    @Singleton
+    public InputTableServiceContextualAuthWiring bindInputTableServiceContextualAuthWiring() {
+        return new InputTableServiceContextualAuthWiring.AllowAll();
+    }
```

Comment:
Why are these services not making it into the map?

Reply:
The Map is for at-rpc-invocation time authorization. The InputTable and the Table services require that the TableReferences be resolved into Tables. The resolution isn't guaranteed until the session runs the export's runnable. The runnable isn't registered until after the rpc has been invoked. So, we can't perform this authorization through the rpc interceptor before the tables have even had a chance to resolve (also, consider complexities regarding batch requests). Instead, these types are deeply injected (i.e. not automatically generated) where the operation has the resolved TableReferences and the original rpc request.